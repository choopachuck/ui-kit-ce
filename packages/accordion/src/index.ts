export * from './Accordion'
export * from './AccordionItem'
export type {
  AccordionIconProps,
  AccordionComponentsConfig,
} from './components'
