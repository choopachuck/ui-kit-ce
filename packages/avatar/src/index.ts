export * from './Avatar'
export * from './AvatarGroup'
export * from './types'
export type {
  AvatarGroupExtraProps,
  AvatarGroupComponentsConfig,
} from './components'
export * from './classes'
export * from './constants'
