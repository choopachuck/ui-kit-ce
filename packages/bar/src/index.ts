export * from './Bar'
export * from './BarButton'
export * from './BarDate'
export * from './BarDivider'
export * from './BarDropdown'
export * from './BarDropdownItem'
export * from './BarItem'
export * from './BarMenuItem'
export * from './BarSearch'
export * from './BarSelect'
export * from './constants'
export * from './interfaces'
export * from './SubBar'
