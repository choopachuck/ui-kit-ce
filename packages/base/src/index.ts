export * from '@v-uik/accordion'
export * from '@v-uik/autocomplete'
export * from '@v-uik/badge'
export * from '@v-uik/bar'
export * from '@v-uik/box'
export * from '@v-uik/breadcrumbs'
export * from '@v-uik/button'
export * from '@v-uik/button-group'
export * from '@v-uik/checkbox'
export * from '@v-uik/checkbox-group'

export * from '@v-uik/combo-box'
export type { Classes as ComboBoxClasses } from '@v-uik/combo-box'
export type { CommonProps as ComboBoxCommonProps } from '@v-uik/combo-box'

export * from '@v-uik/common'
export * from '@v-uik/container'
export * from '@v-uik/date-picker'
export * from '@v-uik/divider'
export * from '@v-uik/drawer'
export * from '@v-uik/dropdown'
export * from '@v-uik/dropdown-menu'
export * from '@v-uik/file-uploader'
export * from '@v-uik/grid'
export * from '@v-uik/hooks'
export * from '@v-uik/input'

export * from '@v-uik/input-helper-text'
export type { Classes as InputHelperTextClasses } from '@v-uik/input-helper-text'

export * from '@v-uik/input-label'
export type { Classes as InputLabelClasses } from '@v-uik/input-label'

export * from '@v-uik/input-number'
export * from '@v-uik/input-password'
export * from '@v-uik/label-control'
export * from '@v-uik/link'
export * from '@v-uik/list'
export * from '@v-uik/masked-input'
export * from '@v-uik/modal'
export * from '@v-uik/notification'
export * from '@v-uik/popup'
export * from '@v-uik/portal'
export * from '@v-uik/progress'
export * from '@v-uik/radio'
export * from '@v-uik/radio-group'

export * from '@v-uik/select'
export type { Classes as SelectClasses } from '@v-uik/select'

export * from '@v-uik/slider'
export type { Classes as SliderClasses } from '@v-uik/slider'

export * from '@v-uik/stepper'
export * from '@v-uik/switch'
export * from '@v-uik/table'
export * from '@v-uik/tabs'

export * from '@v-uik/tag'
export type { CommonProps as TagCommonProps } from '@v-uik/tag'

export * from '@v-uik/textarea'
export * from '@v-uik/theme'

export * from '@v-uik/tooltip'
export type { Classes as TooltipClasses } from '@v-uik/tooltip'

export * from '@v-uik/typography'
export * from '@v-uik/underlay'
export * from '@v-uik/utils'
export * from '@v-uik/labelled'
export * from '@v-uik/avatar'

/**
 * Заглушка, не используется
 * @deprecated
 */
export type Classes = unknown
/**
 * Заглушка, не используется
 * @deprecated
 */
export type CommonProps = unknown
