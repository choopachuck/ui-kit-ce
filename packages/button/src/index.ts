export * from './Button'
export * from './useButtonReset'
