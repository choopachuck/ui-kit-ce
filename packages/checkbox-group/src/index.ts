export * from './CheckboxGroup'
export * from './CheckboxGroupContext'
export * from './interfaces'
