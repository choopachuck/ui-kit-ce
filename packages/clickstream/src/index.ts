export * from './ClickStreamProvider'
export type {
  ClickStreamBaseData,
  ClickStreamContextProps,
  ClickStreamElementProperty,
  ClickStreamElementType,
  ClickStreamEventBatch,
  ClickStreamEventBatchItem,
  ClickStreamEventData,
  ClickStreamEventMeta,
  ClickStreamProviderProps,
} from './types'
