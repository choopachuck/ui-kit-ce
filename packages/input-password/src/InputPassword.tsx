'use client'

import * as React from 'react'
import { Input, InputProps } from '@v-uik/input'
import { useToggle, useButtonReset } from '@v-uik/hooks'
import { HideIcon, ShowIcon } from './Icons'
import { clsx, createUseStyles } from '@v-uik/theme'
import { ButtonProps } from '@v-uik/button'

const useStyles = createUseStyles((theme) => ({
  icon: {
    cursor: 'pointer',
    display: 'flex',
    '&:focus-visible': {
      boxShadow: `0 0 0 2px ${theme.comp.input.colorShadowFocus}`,
    },
  },
  password: {
    fontFamily: theme.comp.inputPassword.typographyFontFamilyHidden,
    fontWeight: theme.comp.inputPassword.typographyFontWeightHidden,
    fontSize: theme.comp.inputPassword.typographyFontSizeHidden,
    lineHeight: theme.comp.inputPassword.typographyLineHeightHidden,
    letterSpacing: theme.comp.inputPassword.typographyLetterSpacingHidden,
  },
}))

type Visibility = boolean

type IconPosition = 'start' | 'end'

export interface InputPasswordProps<TCanClear extends boolean = boolean>
  extends Omit<InputProps<TCanClear>, 'suffix' | 'prefix'> {
  /**
   * Скрыть/показать текст в input'e
   * @default false
   */
  visible?: Visibility
  /**
   * Обработчик нажатия по иконке видимости
   */
  onVisibilityChange?: (value: Visibility) => void
  /**
   * Местоположение иконки: слева или справа
   * @default "end"
   */
  iconPosition?: IconPosition
  /**
   * Иконка, отображаема при скрытом инпуте
   */
  hideIcon?: React.ReactNode
  /**
   * Иконка, отображаема при НЕ скрытом инпуте
   */
  showIcon?: React.ReactNode
  /**
   * Пропсы для кнопки с иконкой
   */
  buttonIconProps?: Omit<ButtonProps, 'className' | 'onMouseDown' | 'onClick'> //TODO: 2.0 ButtonProps оставлены для обратной совместимости, должны быть HTMLProps Button
}

const _InputPassword = React.forwardRef(
  <TCanClear extends boolean = boolean>(
    {
      visible = false,
      onVisibilityChange,
      iconPosition: iconPositionProp = 'end',
      hideIcon: hideIconProp,
      showIcon: showIconProp,
      buttonIconProps = {},
      ...rest
    }: InputPasswordProps<TCanClear>,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const classesList = useStyles()

    const iconPosition = iconPositionProp
    const showIcon = showIconProp || ShowIcon
    const hideIcon = hideIconProp || HideIcon
    const buttonResetClasses = useButtonReset()

    const [toggled, toggle] = useToggle(visible)

    const toggleVisibility = () => {
      toggle()
      onVisibilityChange?.(toggled)
    }

    /*
      Приходится оборачивать в спан, тк нужно навесить курсор и клик
      Тип ReactNode включает в себя еще и строки и числа, а на них
      Не получится навесить ни стили, ни события
    */
    const Icon = (
      // onMouseDown исправляет потерю фокуса с input
      <button
        type="button"
        className={clsx(buttonResetClasses.resetButton, classesList.icon)}
        onClick={toggleVisibility}
        onMouseDown={(e) => e.preventDefault()}
        {...buttonIconProps}
      >
        {toggled ? showIcon : hideIcon}
      </button>
    )

    const classes = {
      ...rest.classes,
      input: clsx(rest.classes?.input, {
        [classesList.password]: !toggled,
      }),
    }

    return (
      <Input
        {...rest}
        ref={ref}
        suffix={iconPosition === 'end' && Icon}
        prefix={iconPosition === 'start' && Icon}
        classes={classes}
        inputProps={{
          ...rest.inputProps,
          type: toggled ? 'input' : 'password',
        }}
      />
    )
  }
)

export const InputPassword = _InputPassword as <
  TCanClear extends boolean = boolean
>(
  props: InputPasswordProps<TCanClear> & {
    ref?: React.ForwardedRef<HTMLDivElement>
  }
) => React.ReactElement
