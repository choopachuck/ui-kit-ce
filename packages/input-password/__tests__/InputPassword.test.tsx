import * as React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { InputPassword } from '../src'

it(' can work as an input', () => {
  const onChange = jest.fn()
  const { getByDisplayValue } = render(
    <InputPassword value="text" onChange={onChange} />
  )
  expect(onChange).toHaveBeenCalledTimes(0)
  fireEvent.change(getByDisplayValue('text'), {
    target: { value: 'new value' },
  })
  expect(onChange).toHaveBeenCalledTimes(1)
  expect(onChange).toHaveBeenCalledWith('new value', expect.anything(), 'input')
})

it(' has the option of moving the icon to the left', () => {
  const { rerender, container } = render(<InputPassword />)
  expect(container).toMatchSnapshot()
  rerender(<InputPassword iconPosition="start" />)
  expect(container).toMatchSnapshot()
})

it(' could take text as an icon', () => {
  const { queryByText, getByText } = render(
    <InputPassword hideIcon="111" showIcon="222" />
  )
  expect(queryByText('111')).toBeInTheDocument()
  expect(queryByText('222')).not.toBeInTheDocument()

  fireEvent.click(getByText('111'))
  expect(queryByText('111')).not.toBeInTheDocument()
  expect(queryByText('222')).toBeInTheDocument()
})

it('could take component as an icon', () => {
  const hideIcon = <span>hideIcon</span>
  const showIcon = <span>showIcon</span>
  const { queryByText, getByText } = render(
    <InputPassword hideIcon={hideIcon} showIcon={showIcon} />
  )
  expect(queryByText('hideIcon')).toBeInTheDocument()
  expect(queryByText('showIcon')).not.toBeInTheDocument()

  fireEvent.click(getByText('hideIcon'))
  expect(queryByText('hideIcon')).not.toBeInTheDocument()
  expect(queryByText('showIcon')).toBeInTheDocument()
})

it('handle clear value correctly', () => {
  const handleChange = jest.fn()
  const inputValue = 'new value'
  const { getByRole, rerender, container } = render(
    <InputPassword canClear value="" onChange={handleChange} />
  )
  fireEvent.change(
    container.querySelector('input[type="password"]') as Element,
    { target: { value: inputValue } }
  )
  expect(handleChange).toHaveBeenCalledTimes(1)
  expect(handleChange).toHaveBeenCalledWith(
    inputValue,
    expect.anything(),
    'input'
  )

  rerender(
    <InputPassword canClear value={inputValue} onChange={handleChange} />
  )

  const clearButton = getByRole('button', { name: 'clearButton' })
  expect(clearButton).toBeInTheDocument()
  fireEvent.mouseDown(clearButton)

  expect(handleChange).toHaveBeenCalledTimes(2)
  expect(handleChange).toHaveBeenCalledWith('', expect.anything(), 'clear')
})
