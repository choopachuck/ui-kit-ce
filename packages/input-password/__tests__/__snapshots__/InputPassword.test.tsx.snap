// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` has the option of moving the icon to the left 1`] = `
<div>
  <div
    class="container-0-2-4"
  >
    <div
      class="root-0-2-22 medium-0-2-28"
    >
      <input
        class="input-0-2-30 password-0-2-2 inputMedium-0-2-32"
        type="password"
        value=""
      />
      <div
        class="affix-0-2-38 suffix-0-2-35 suffix-0-2-41"
      >
        <button
          class="resetButton-0-2-3 icon-0-2-1"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M2.937 3.36612C2.44885 3.85427 2.44885 4.64572 2.937 5.13388L5.93092 8.1278C4.69525 9.00632 3.46629 10.1547 2.24816 11.573L2.23942 11.583C1.91318 11.9632 1.92444 12.5267 2.24816 12.8929C6.60756 17.9686 11.1057 19.5878 15.5536 17.7505L18.8469 21.0438C19.335 21.5319 20.1265 21.5319 20.6146 21.0438C21.1028 20.5556 21.1028 19.7641 20.6146 19.276L17.8379 16.4993C19.1515 15.6017 20.4578 14.3996 21.7519 12.8929C22.0756 12.5267 22.0868 11.9633 21.7606 11.583L21.7519 11.573C17.3028 6.39284 12.7092 4.81293 8.17192 6.83327L4.70476 3.36612C4.21661 2.87796 3.42515 2.87796 2.937 3.36612ZM7.62094 9.81782C7.22525 10.5338 7.00003 11.3571 7.00003 12.2329C7.00003 14.9944 9.2386 17.2329 12 17.2329C12.8759 17.2329 13.6992 17.0077 14.4151 16.612L12.4884 14.6853C12.3304 14.7165 12.1671 14.7329 12 14.7329C10.6193 14.7329 9.50003 13.6137 9.50003 12.2329C9.50003 12.0658 9.51643 11.9025 9.5477 11.7446L7.62094 9.81782ZM11.2018 9.86311L9.33824 7.99959C10.109 7.51393 11.0217 7.23296 12 7.23296C14.7614 7.23296 17 9.47153 17 12.2329C17 13.2113 16.719 14.124 16.2334 14.8947L14.3699 13.0312C14.4543 12.7805 14.5 12.5121 14.5 12.2329C14.5 10.8522 13.3807 9.73295 12 9.73295C11.7209 9.73295 11.4524 9.7787 11.2018 9.86311Z"
              fill="black"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[` has the option of moving the icon to the left 2`] = `
<div>
  <div
    class="container-0-2-4"
  >
    <div
      class="root-0-2-22 medium-0-2-28"
    >
      <div
        class="affix-0-2-38 prefix-0-2-34 prefix-0-2-40"
      >
        <button
          class="resetButton-0-2-3 icon-0-2-1"
          type="button"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M2.937 3.36612C2.44885 3.85427 2.44885 4.64572 2.937 5.13388L5.93092 8.1278C4.69525 9.00632 3.46629 10.1547 2.24816 11.573L2.23942 11.583C1.91318 11.9632 1.92444 12.5267 2.24816 12.8929C6.60756 17.9686 11.1057 19.5878 15.5536 17.7505L18.8469 21.0438C19.335 21.5319 20.1265 21.5319 20.6146 21.0438C21.1028 20.5556 21.1028 19.7641 20.6146 19.276L17.8379 16.4993C19.1515 15.6017 20.4578 14.3996 21.7519 12.8929C22.0756 12.5267 22.0868 11.9633 21.7606 11.583L21.7519 11.573C17.3028 6.39284 12.7092 4.81293 8.17192 6.83327L4.70476 3.36612C4.21661 2.87796 3.42515 2.87796 2.937 3.36612ZM7.62094 9.81782C7.22525 10.5338 7.00003 11.3571 7.00003 12.2329C7.00003 14.9944 9.2386 17.2329 12 17.2329C12.8759 17.2329 13.6992 17.0077 14.4151 16.612L12.4884 14.6853C12.3304 14.7165 12.1671 14.7329 12 14.7329C10.6193 14.7329 9.50003 13.6137 9.50003 12.2329C9.50003 12.0658 9.51643 11.9025 9.5477 11.7446L7.62094 9.81782ZM11.2018 9.86311L9.33824 7.99959C10.109 7.51393 11.0217 7.23296 12 7.23296C14.7614 7.23296 17 9.47153 17 12.2329C17 13.2113 16.719 14.124 16.2334 14.8947L14.3699 13.0312C14.4543 12.7805 14.5 12.5121 14.5 12.2329C14.5 10.8522 13.3807 9.73295 12 9.73295C11.7209 9.73295 11.4524 9.7787 11.2018 9.86311Z"
              fill="black"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
      <input
        class="input-0-2-30 password-0-2-2 inputMedium-0-2-32"
        type="password"
        value=""
      />
    </div>
  </div>
</div>
`;
